daprdocs/content/en/developing-applications/building-blocks/state-management/howto-state-query-api.md

```diff
@@ -0,0 +1,140 @@
+---
+type: docs
+title: "How-To: Query the state"
+linkTitle: "How-To: Query the state"
+weight: 250
+description: "API for querying state stores"
+---
+
+{{% alert title="alpha" color="warning" %}}
+The state query API is in **alpha** stage.
+{{% /alert %}}
+
+## Introduction
+
+The state query API provides a way of querying the key/value data stored in state store components. This query API is not a replacement for a complete query language, and is focused on retrieving, filtering and sorting key/value data that you have saved through the state management APIs.
+
+This query API does not support querying of actor state stored in a state store. For that you need to use the query API for the specific database.
+See [querying actor state]({{< ref "state-management-overview.md#querying-actor-state" >}}).
+
+You can find additional information in the [resources]({{< ref "#resources" >}}) section.
+
+## Querying the state
+
+You submit query requests via HTTP POST/PUT or gRPC.
+The body of the request is the JSON map with 3 entries: `filter`, `sort`, and `pagination`.
+
+The `filter` section specifies the query conditions in the form of a tree of key/value operations, where the key is the operator and the value is the operands.
+
+The following operations are supported:
+| Operator | Operands | Description |
+|----------|-----------|-------------|
+| `EQ`       | key:value  | key == value|
+| `IN`        | key:[]value| key == value[0] OR key == value[1] OR ... OR key == value[n] |
+| `AND`    | []operation|  operation[0] AND operation[1] AND ... AND operation[n] |
+| `OR`      | []operation|  operation[0] OR operation[1] OR ... OR operation[n] |
+
+The `sort` is an optional section is an ordered array of `key:order` pairs, where `key` is a key in the state store, and the `order` is an optional string indicating sorting order: `"ASC"` for ascending and `"DESC"` for descending. If omitted, ascending order is the default.
+
+The `pagination` is an optional section containing `limit` and `token` parameters. `limit` sets the page size. `token` is an iteration token returned by the component, and used in consecutive queries.
+
+Below is an example of the query request JSON body.
```

Comment:
There are two things you need to add here
1) Have the example data in a table that you show people that you are going to query. In other words have 3-5 rows of data stored as key/value pairs that people can really see that this is stored like. And then say these are records of people with information about they work roles and where they work
2) Then start with a really simple query example. Use only the `sort` operator, so get a list of people names so that people see this with a "hello world" like example on the data above.
3) Then add a runable bash and Powershell script like the example here https://docs.dapr.io/developing-applications/building-blocks/state-management/howto-get-save-state/ to show how this query would be used in practice. 
4) Ideally you would also add a code example also in Python.
5) Then below  you can then say, "Now let's look at a more complex query example"

The whole idea here is that you need to explain this to someone brand new and it really helps to see the full picture. Currently I am trying to figure out what this data key/value format looks like and the full walk through of the data that I originally stored would help. In fact now I think about this, I would suggest adding a Save bash/powershell script example for one of the person records, so that someone could try this out on a Dapr sidecar instance themselves. That would then make the how to more complete, so 
1) Bash/Powershell to save person record example
2)  Bash/Powershell  to query with a simple and more complex example.